pdns/proxy-protocol.cc

```diff
@@ -82,8 +82,7 @@ std::string makeProxyHeader(bool tcp, const ComboAddress& source, const ComboAdd
 
   const uint16_t contentlen = htons(static_cast<uint16_t>(total));
   std::string ret;
-  ret.reserve(total);
-  makeSimpleHeader(command, protocol, contentlen, ret);
+  makeSimpleHeader(command, protocol, contentlen, total, ret);
```

Comment:
Maybe just a wording issue, but it looks like you are passing the `total` here, and the allocation in `makeSimpleHeader()`  is expecting an additional count

Reply:
You are right, something does not look correct at all. I'll re-check everything, thanks!

Reply:
I just pushed a fix, thanks for spotting this!